src/controller/expensesController.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+
+const { ExpensesService } = require('../services/expenses');
+const { usersController } = require('./usersController');
+
+class ExpensesController extends ExpensesService {
+  constructor() {
+    super();
+
+    this.postExpense = this.postExpense.bind(this);
+    this.createExpense = this.createExpense.bind(this);
+    this.getExpenses = this.getExpenses.bind(this);
+    this.getAll = this.getAll.bind(this);
+    this.getExpense = this.getExpense.bind(this);
+    this.getOne = this.getOne.bind(this);
+    this.removeExpense = this.removeExpense.bind(this);
+    this.removeOne = this.removeOne.bind(this);
+    this.patchExpense = this.patchExpense.bind(this);
+    this.modifyExpence = this.modifyExpence.bind(this);
+  }
+
+  postExpense(req, res) {
+    const { userId, spentAt, title, amount, category, note } = req.body;
+
+    const users = usersController.getAll();
+    const hasUser = users.find(user => +userId === user.id);
+
+    const isValid = hasUser && spentAt && title && amount && category && note;
+
+    const error = hasUser
+      ? 'required parameters is not passed'
+      : 'user does\'nt exist';
+
+    if (!isValid) {
+      res.status(400);
+      res.json({ error });
+
+      return;
+    }
+
+    const expense = super.createExpense({
+      userId, spentAt, title, amount, category, note,
+    });
+
+    res.statusCode = 201;
+    res.json(expense);
+  }
+
+  getExpenses(req, res) {
+    const expenses = super.getAll();
+
+    const { userId, category, to, from } = req.query;
+
+    if (from && to) {
+      const expensesByDate = expenses.filter(
+        (expense) => expense.spentAt > from && expense.spentAt < to
+      );
+
+      res.statusCode = 200;
+      res.json(expensesByDate);
+
+      return;
+    }
+
+    if (userId && category) {
```

Comment:
Category can be array, check api docs.

Reply:
it cann't

Reply:
I was using strict data schema from api docs, and here category was only string 
![image](https://user-images.githubusercontent.com/105124305/204472633-7e4c27de-a7ab-443b-8d33-90a29475e66e.png)

Reply:
You are showing me `PATCH / expenses/:id` method, but I've commented you about `GET /expenses` method. Look closely. `Category` (not `categories` - it's incorrect) can be array. F.e.: 

```http
GET /expenses?userId=1&category=Food&category=Sport
```

You need to filter and include both categories. It's not present in tests but [described here](https://app.swaggerhub.com/apis/YURIIHOLIUK_1/Accounting-App/1.0.0#/Expenses/getExpenses).